### PR TITLE
test_caless: add SAN dNSName extensions for wildcard tests

### PR DESCRIPTION
**Pull request by Rezney**
It may happen that FQDN does not match with the domain mapped to
the host. In this case we add wildcard for both domains.

### CODE REVIEW

**Comment by Rezney** [comment]
E.g. in a lab environment there may be hostname vm-007.ipa.test and domain dom-007.ipa.test.

File: ipatests/test_integration/create_caless_pki.py [hunk 1]
```diff
@@ -204,6 +204,21 @@ def profile_server(builder, ca_nick, ca,
...
+                     x509.DNSName(u'*.' + domain),
+                     x509.DNSName(u'*.' + server_domain),
+                ]),
+                critical=False,
+            )
+        else:
+            builder = builder.add_extension(
+                x509.SubjectAlternativeName([x509.DNSName(u'*.' + domain)]),
+                critical=False,
+            )
```
Comment (frasertweedale) [inline ipatests/test_integration/create_caless_pki.py]: I think the logic is fine, but a few comments:

1. partially build the GeneralName list, then append to it if the condition matches (eliminate duplicate code)
2. if for some reason there is no `.` in `domain`, we don't want to crash when indexing the split result at `[1]`.

So, something like:

```python
if wildcard:
  names = [x509.DNSName(u'*.' + domain)]
  server_split = server1.split('.', 1)
  if len(server_split) == 2 and domain != server_split[1]
    names.append(x509.DNSName(u'*.' + server_split[1])
  builder = builder.add_extension(x509.SubjectAlternativeName(names), critical=False)
```

**Comment by frasertweedale** [comment]
Logic looks fine, code looks fine.  I haven't tested it but it's obviously not getting
in the way or regressing anything else, so ACK.

**Comment by tomaskrizek** [comment]
master:

* a3c99367bfe1071073cd93237660d783459b25e2 test_caless: add SAN dNSName extensions for wildcard tests

**Pull request closed without merging**